mod-source-record-manager-server/src/main/java/org/folio/services/ChangeEngineServiceImpl.java

```diff
@@ -338,6 +348,25 @@ public List<Record> getParsedRecordsFromInitialRecords(List<InitialRecord> rawRe
       }).collect(Collectors.toList());
   }
 
+  private ParsedResult validateIf999ffFieldExistsOnInstanceCreateAction(JobExecution jobExecution, ParsedResult parsedResult) {
+    if (jobExecution.getJobProfileInfo().getDataType().equals(DataType.MARC)) {
+      var tmpRecord = new Record()
+        .withParsedRecord(new ParsedRecord().withContent(parsedResult.getParsedRecord().encode()));
+      if ((StringUtils.isNotBlank(getValue(tmpRecord, TAG_999, SUBFIELD_S)) && hasIndicator(tmpRecord, SUBFIELD_S))
+        || (StringUtils.isNotBlank(getValue(tmpRecord, TAG_999, SUBFIELD_I)) && hasIndicator(tmpRecord, SUBFIELD_I))
+        && createInstanceActionExists(jobExecution)) {
+        ParsedResult result = new ParsedResult();
+        JsonObject errorObject = new JsonObject();
+        errorObject.put("error", INSTANCE_CREATION_999_ERROR_MESSAGE);
+        result.setErrors(errorObject);
+        result.setHasError(true);
```

Comment:
This is not your problem. The hasError field should be automatically set when an errors object is set.

Reply:
Hi @okolawole-ebsco! Agree. I changed method "isHasError" inside ParsedResult.